src/main/java/ir/sahab/cleanup/Cleanups.java

```diff
@@ -68,16 +65,25 @@ public void doAllQuietly() {
 
     public void doAll() throws IOException {
         boolean allSucceeded = true;
-        for (RunnableWithException<Exception> closeStatement : closeStatements) {
+        // Probably the first error was the cause of the problem for the rest of the operations and therefore
+        // the most important error to investigate the problem. For this reason, only the first error is
+        // returned after the entire clean-up operations.
+        IOException firstException = null;
+        for (AutoCloseable closeStatement : closeStatements) {
             try {
-                closeStatement.run();
+                closeStatement.close();
             } catch (Exception e) {
                 logger.error("Failed to run clean-up statement.", e);
+                if (firstException == null) {
+                    // Most of te time, the {@code close()} function in which {@link IOException} occurs is used.
+                    // For this reason, we put the catched error in it.
+                    firstException = new IOException("Failed to clean-up all resources.", e);
```

Comment:
It is better to assign the firstException with just e.
Then when you want to throw it up, wrap it in an IOException.